### DISCUSSION TITLE
MSI instead of EXE for elastic agent

### DISCUSSION THREAD

**Discussion by MDCAllan** [Security-Onion-Solutions/securityonion#11993] [Ideas] [answered]
On SO 2.3 the various windows agent options (winlogbeat, wazuh, and osquery) all downlaoded as MSI files, which are very easy to deploy and uninstall in a managed environment, as they contain and install key information about how to see if the software is installed an how to remove it. I can build an Intune deployment for these and orchestrate the install or removal of any of them very easily throughout the environment. On our current deployment, we're using winlogbeat and I push a custom config file and encryption certificates/keys to encrypt the payload with group policy and a robocopy command script. Works great. When it comes time to move all the computers to our 2.4 SO deployment (running concurrently), I can easily "clean up" the old agent/s since they were installed via MSI file. 

SO 2.4 deploys the new combo-agent (so-elastic-agent_windows_amd64) as an EXE. I don't see any uninstaller included the install. I'm fearful of deploying this with no clear automated way to uninstall from hundreds of computers in the future. 

Can you have it package the elastic agent / osquery combo agent as an MSI instead of EXE? 

Thanks!

**Comment by robbiemarshall** [comment, marked as answer]
Elastic has not published a MSI, and does not appear to be planning on doing it anytime soon
The Elastic Agent can be cleanly uninstalled with `elastic-agent uninstall`
https://www.elastic.co/guide/en/fleet/current/uninstall-elastic-agent.html
